File: backend/src/main/java/cloudflight/integra/backend/model/Activity.java
package cloudflight.integra.backend.model;

import cloudflight.integra.backend.model.utils.enums.ActivityType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Data
@Table(name = "activities")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Activity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ActivityType type;

    @Column(nullable = false)
    private UUID targetId;

    @Column(nullable = false)
    private String targetType; 

    @Column
    private String targetName;

    @Column
    private String targetData;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/app/shared/models/activity/Activity.ts
export type ActivityType = 'REVIEW_CREATED' | 'POI_CREATED' | 'QUEST_COMPLETED';

export interface ActivityDto {
  id: string;
  userId: string;
  username: string;
  userAvatarUrl?: string;
  activityType: ActivityType;
  actionDescription: string;
  targetId: string;
  targetType: string;
  targetName: string;
  targetData?: string;
  createdAt: string;
}
